test/opt/eliminate_dead_functions_test.cpp

```diff
@@ -0,0 +1,189 @@
+// Copyright (c) 2017 Google Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <string>
+#include <vector>
+
+#include <gmock/gmock.h>
+
+#include "assembly_builder.h"
+#include "pass_fixture.h"
+#include "pass_utils.h"
+
+namespace {
+
+using namespace spvtools;
+using ::testing::HasSubstr;
+
+using EliminateDeadFunctionsBasicTest = PassTest<::testing::Test>;
+
```

Comment:
Please add a test prove that when you remove a function, you also remove decoration instructions that target IDs inside that function.

E.g. try this GLSL shader:
#version 450

precision lowp float;
precision highp int;
void main() {
  lowp float x = 1.0;
  lowp float y = 1.0;
  lowp float z =  x + y;
}

This will generate OpVariable and OpFAdd inside the function, and there are OpDecorate ... RelaxedPrecision targeting those IDs.
(You'll have to artificially create a dead function to wrap around them since glslang only emits reachable functions.)